src/platform/libretro/libretro.c

```diff
@@ -888,10 +912,10 @@ bool retro_load_game(const struct retro_game_info* game) {
 	memset(outputBuffer, 0xFF, size);
 	core->setVideoBuffer(core, outputBuffer, 256);
 
-	core->setAudioBufferSize(core, SAMPLES);
-
-	blip_set_rates(core->getAudioChannel(core, 0), core->frequency(core), 32768);
-	blip_set_rates(core->getAudioChannel(core, 1), core->frequency(core), 32768);
+	_loadAudioSampleRate();
+	core->setAudioBufferSize(core, 1);
```

Comment:
This isn't going to work very well.

Reply:
I tried it with both GB/C and GBA games and thought it worked great for me, with better frontend sync timing results.

If you refer to worse speed performance (?), I can change it back to SAMPLES if that'd be preferable. Or I am missing something else important??

Reply:
Sorry, I had a terrible headache so I couldn't get in a good reply earlier, but basically the entire premise of this PR is flawed: increasing the sample rate over 32768 will actually make the audio quality *worse*. The GBA inherently samples at 32768 samples per second, and all this PR will do will resample that in the backend, leading to slight audio quality degradation. Toying with the buffer's sample rate won't affect how quickly the emulator runs in any capacity.

This specific line is bad because it sets the audio buffer to be a single entry, which, given the fact that you've set it up in a way where a single sample can create multiple entries, is effectively undefined behavior and will likely break in unexpected ways. Moreover, it'll slow down the emulator because it has to process audio samples far more often. It can also break dynamic rate control, I think, which is one of RetroArch's key features.

The right way to do this is to have the emulator output audio at its native rate, and have the frontend resample it, which I believe is how it was set up before.

Reply:
I understand now. Thank you for clearly explaining!

I will close but have 1 more question. The internal gb(c) chip tune runs at a very high rate. Would this lead to audio degradation when running that system also? SameBoy outputs to frontend at 384000 which is the root source of my damaged tinkering.

Reply:
mGBA does not support higher output rates for GB audio at this time, though I may change that in the future.